pandas/tests/arrays/string_/test_string.py

```diff
@@ -556,10 +549,10 @@ def test_arrow_load_from_zero_chunks(
 
 
 def test_value_counts_na(dtype):
-    if getattr(dtype, "storage", "") == "pyarrow":
-        exp_dtype = "int64[pyarrow]"
-    elif getattr(dtype, "storage", "") == "pyarrow_numpy":
+    if dtype.na_value is np.nan:
         exp_dtype = "int64"
+    elif dtype.storage == "pyarrow":
+        exp_dtype = "int64[pyarrow]"
```

Comment:
Somewhat surprised by this behavior - so depending on if pyarrow is installed or not you will get back two different types? Is it not possible to just always return pd.Int64Dtype()

Reply:
Yeah, I am personally not a fan of this behaviour (and if I knew we were changing this I think I would have objected). This seems to have changed somewhere in the 2.x releases:

With pandas 1.5
```
In [3]: pd.Series(["a", "b", "a"], dtype="string[python]").value_counts()
Out[3]: 
a    2
b    1
dtype: Int64

In [4]: pd.Series(["a", "b", "a"], dtype="string[pyarrow]").value_counts()
Out[4]: 
a    2
b    1
dtype: Int64
```

With pandas 2.2:

```
In [1]: pd.Series(["a", "b", "a"], dtype="string[pyarrow]").value_counts()
Out[1]: 
a    2
b    1
Name: count, dtype: int64[pyarrow]

In [2]: pd.Series(["a", "b", "a"], dtype="string[python]").value_counts()
Out[2]: 
a    2
b    1
Name: count, dtype: Int64
```

But so this is not introduced by this PR (I just reordered the clauses here, seemingly giving a bigger diff), and given this is strictly not related and potentially contentious to change, it's definitely a topic for a different issue/PR :)

Reply:
Fair enough on this PR, though this behavior is a little stranger now that we have one `pd.StringDtype` class instead of separate python / arrow types. This leaks implementation details into the API space

Diff suggests it started with https://github.com/pandas-dev/pandas/pull/51542 so cc @mroeschke for any insights, but I would really like to move away from this behavior in follow ups

Reply:
> though this behavior is a little stranger now that we have one `pd.StringDtype` class instead of separate python / arrow types

To be clear, that's something we already have for quite a time. This PR is only touching how the NaN-variant of `StringDtype` is constructed, it does not touch the NA-variants of the `StringDtype` (and the fact that we also have a `ArrowDtype("string")` ..). The behaviour you (rightfully IMO) bring up is only for the NA-variant.

Reply:
Ah ok makes sense - misinterpreted this as changing the NA variant. 

Still a bit strange that the NA and NaN variant return different types, but not something to solve here

Reply:
> Still a bit strange that the NA and NaN variant return different types, but not something to solve here

That part is then actually intentionally (although also not changed in this PR), as that is one of the essential parts of the PDEP, quoting from the section about missing value semantics:

> In practice, this means that the default string dtype will use `NaN` as
> the missing value sentinel, and:
>
> - String columns will follow NaN-semantics for missing values, where `NaN` gives
>   False in boolean operations such as comparisons or predicates.
> - Operations on the string column that give a numeric or boolean result will use
>   the default data types (i.e. numpy `int64`/`float64`/`bool`).

Reply:
Actually need to take another look at this. I still think pd.StringDtype + pd.NA should always return a pd.Int64Dtype here regardless of if pyarrow is installed or not

Reply:
> Actually need to take another look at this. I still think pd.StringDtype + pd.NA should always return a pd.Int64Dtype here regardless of if pyarrow is installed or not

Can you open a new issue about that to discuss this? 
As mentioned above, this is both not affected by as not really related to this PR (this PR only deals with the NaN variant, not the NA variant of the dtype, and the behaviour you question is solely related to the NA variant).

Reply:
Ah OK - I did not realize this was already in main. Opened up https://github.com/pandas-dev/pandas/issues/59346 for discussion; hoping we can revert that behavior before too long